### ISSUE TITLE
Add `--rm` as an option for `pulumi destroy`

### ISSUE THREAD

**Issue by cnunciato** [pulumi/pulumi#10484]
I find myself doing this a lot, particularly when I'm working with multiple stacks in multiple projects:

```
pulumi destroy --yes && pulumi stack rm --yes && pulumi up --yes
```

This works, but I often forget the `rm` part, which leads to errors and/or orphaned stacks that pile up in the Pulumi Service and cause problems for me later. It'd be nice if I could pass a `--rm` option to `pulumi destroy` to have the stack destroyed and removed in one go:

```
pulumi destroy --rm
```

The Docker CLI does something similar, as it's so common to want to run and immediately remove a container:

```
docker run some-thing -it --rm 
``` 

I'm not sure how we'd want this to interact with `--yes` (does `--rm` imply it? maybe? kind of like `--skip-preview` does for `up`?), but in general it seems like a minor addition that might make some users (starting with this one! 😄) happy.

**Comment by susanev** [comment]
from the usability studies, there was some general confusion related to destroy vs rm, so def support this idea!

**Comment by dixler** [comment]
added to pulumi/home#1893 backlog

**Comment by dixler** [comment]
> This works, but I often forget the rm part, which leads to errors and/or orphaned stacks that pile up in the Pulumi Service and cause problems for me later.

Might the service hide destroyed stacks to make it easier?

**Comment by cnunciato** [comment]
> Might the service hide destroyed stacks to make it easier?

Perhaps, but the pain that I mainly run into with this is with projects that already exist in the service when I'm using the CLI (i.e., project-name conflicts). So if I had to take an extra step to show them before deleting them, it might even be a bit more frustrating.

**Comment by shubbham1215** [comment]
I would like to contribute and work on this issue. Can you please assign to me?

**Comment by dixler** [comment]
Thanks for the interest. Going to discuss around this a little more, will get back to you!

**Comment by dixler** [comment]
Hey, @shubbham1215 there seems to be a lot of interest around this and your PR would be greatly appreciated!

**Comment by RobbieMcKinstry** [comment]
Hey @shubbham1215 just wanted to check in. Are you still interested in picking up a PR for this feature? Thanks :)

**Comment by shubbham1215** [comment]
@RobbieMcKinstry Isn't this already merged?I can see  pull request  https://github.com/pulumi/pulumi/pull/10943 for this issue that shows status as merged.

**Comment by RobbieMcKinstry** [comment]
You're right! I didn't see that, since this issue wasn't linked. I'm going to close this issue since the feature is now in :) 

Thanks much friend!
